loadtest/publish_short_requests.py

```diff
@@ -0,0 +1,183 @@
+
+_API = None
+
+# This is the server endpoint to run the load tests against.
+# Please uncomment the right line before running this file or define it with
+# the environment variable API_ENDPOINT.
+# We don't specify a default server so that the user has to consciously decide
+# the server to hit.
+
+# This is your local server.
+# _API = 'http://localhost:5252'
+
+# This is the staging server.
+# _API = 'https://dev.firefoxprofiler.nonprod.cloudops.mozgcp.net'
+
+# This is the various file sizes we'll generate in the global setup.
+_FILE_SIZES = (1, 10, 50)
+
+
+def setup_api_endpoint():
+    """Sets up the _API global that we use in all scenarii.
+    """
+    global _API
+
+    endpoint_from_env = os.getenv('API_ENDPOINT')
+    if endpoint_from_env is not None:
+        # The environment always wins over what's defined in the file.
+        _API = endpoint_from_env
+
+    if _API is None:
+        raise ValueError(
+                'Please define one API endpoint before running the test file, '
+                'either uncommenting the right line in the file or specifying '
+                'the env variable API_ENDPOINT.'
+                )
+    print(f"We'll run tests against the endpoint {_API}.")
+
+
+@global_setup()
+def test_starts(args):
+    """ This functions is called before anything starts.
+
+    Notice that it's not a coroutine.
+
+    Here we configure various things:
+    * we configure the API endpoint we'll use in the tests.
+    * we generate the various files to be sent in the tests.
+    """
+    setup_api_endpoint()
+    files = {x: os.urandom(x * 1024) for x in _FILE_SIZES}
+    set_var("files", files)
+
+
+@setup()
+async def worker_starts(worker_id, args):
+    """ This function is called once per worker.
+
+    If it returns a mapping, it will be used with all requests.
+
+    You can add things like Authorization headers for instance,
+    by setting a "headers" key.
+    """
+    headers = dict(Accept='application/vnd.firefox-profiler+json;version=1')
+    return dict(headers=headers)
+
+
+@teardown()
+def worker_ends(worker_id):
+    """ This functions is called when the worker is done.
+
+    Notice that it's not a coroutine.
+    """
+    pass
+
+
+@global_teardown()
+def test_ends():
+    """ This functions is called when everything is done.
+
+    Notice that it's not a coroutine.
+    """
+    pass
+
+
+# This is copied from decode_jwt_payload.py
+def get_profile_token_from_jwt(jwt):
+    """This handles one line of input, that should be a JWT token.
+
+    This will first split the token in its 3 components, base64 decode the 2nd
+    one that's the payload, json-parse it, and finally print the key
+    "profileToken" from that JSON payload.
+    """
+    _header, payload, _signature = jwt.strip().split(".")
+
+    decoded_str = jwt_base64_decode(payload)
+    json_payload = json.loads(decoded_str)
+    token = json_payload['profileToken']
+    return token
+
+
+def jwt_base64_decode(payload):
```

Comment:
Optional: you should be able to collect these in a utility module.

Reply:
Yeah, I wrote this second file very quickly by copy pasting... I believe you're right :-)

Reply:
I collected them in a separate file, but I don't want to use it in `decode_jwt_payload.py` because I want that file to be standalone so that it can be copy pasted to use it elsewhere.

Reply:
plot twist: this doesn't work with molotov right now (the module isn't found) if I don't transform all this in a package. This is more work than I want to do now. We'll do the work if we need this more later.